cr.c

```diff
@@ -88,14 +95,32 @@ int dill_ctx_cr_init(struct dill_ctx_cr *ctx) {
     ctx->r = &ctx->main;
     dill_qlist_init(&ctx->ready);
     dill_rbtree_init(&ctx->timers);
-    ctx->last_poll = now();
     /* Initialize the main coroutine. */
     memset(&ctx->main, 0, sizeof(ctx->main));
     ctx->main.ready.next = NULL;
     dill_slist_init(&ctx->main.clauses);
 #if defined DILL_CENSUS
     dill_slist_init(&ctx->census);
 #endif
+    /* Initialize the external event poll timer for 1 second intervals. */
+    struct sigaction sa;
+    sa.sa_flags = SA_SIGINFO;
+    sa.sa_sigaction = dill_ctx_timer_handler;
+    sigemptyset(&sa.sa_mask);
+    if (sigaction(SIGRTMIN, &sa, NULL) == -1) return 1;
+    struct sigevent sev;
+    sev.sigev_notify = SIGEV_SIGNAL;
+    sev.sigev_signo = SIGRTMIN;
```

Comment:
Should we make this configurable somehow? SIGRTMIN to SIGRTMAX is available to the application for use.

On OSX, we have no choice but to use SIGALRM.  Maybe just use `alarm` on all platforms?